src/vm/analysis/check_typing/mod.rs

```diff
@@ -138,9 +131,9 @@ fn check_atomic_type(atom: AtomTypeIdentifier, to_check: &TypeSignature) -> Chec
     }
 }
 
-impl <'a, 'b> TypeChecker <'a, 'b> {
-    fn new(db: &'a AnalysisDatabase<'b>) -> TypeChecker<'a, 'b> {
-        TypeChecker {
+impl <'a, 'b> CheckTyping <'a, 'b> {
```

Comment:
I'm confused -- is there a reason the name of this struct has changed from `TypeChecker` to `CheckTyping`?

Reply:
I have the same question about the name of the module -- what is meant by `check_typing`, and how is it different from `typechecker`?  Would it be possible to come up with a clearer name -- preferably a noun?